packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -288,8 +295,12 @@ function useFocusContainment(scopeRef: RefObject<HTMLElement[]>, contain: boolea
         // Use document.activeElement instead of e.relatedTarget so we can tell if user clicked into iframe
         if (scopeRef === activeScope && !isElementInChildScope(document.activeElement, scopeRef)) {
           activeScope = scopeRef;
-          focusedNode.current = e.target;
-          focusedNode.current.focus();
+          if (document.body.contains(e.target)) {
+            focusedNode.current = e.target;
+            focusedNode.current.focus();
+          } else if (activeScope) {
+            focusFirstInScope(activeScope.current, !allowFocusableFirstInScope);
```

Comment:
In the case of the Dialog (and perhaps other components that have the same kind of issue occur), should an effort be made to focus any tabbable content first and then fall back to any focusable but non tabbable content? I would've been confused as to where focus went in the [test example](https://reactspectrum.blob.core.windows.net/reactspectrum/b76123c5765d9482965ba48288496733ff2a8a39/storybook/index.html?path=/story/focusscope--allow-focusable-first-in-scope) if it didn't have a focus outline, something which our RSP Dialogs don't have I think.

I'm not sure what the expected behavior should be here, mainly speaking from a UX perspective so feel free to correct me if the current behavior is correct. I can see why we may want to do this (i.e. allows the Dialog aria-labelledby to be announced) so I'm fine with the behavior as is as well

Reply:
If we just call `focusFirstInScope(activeScope.current)` the first tabbable element within scope would get focus, and we wouldn't necessarily need an API change adding `allowFocusableFirstInScope`. I added the prop, so the Dialog, which has `tabIndex="-1"` would be able to accept focus, like [the default behavior when the Dialog opens](https://github.com/adobe/react-spectrum/blob/7e8a7f69685139b3574453b424a5ad4c8fa4e808/packages/%40react-aria/dialog/src/useDialog.ts#L35-L54),  when the contents of the FocusScope are replaced. I think focusing the Dialog itself is in some ways preferable to focusing the first tabbable descendant, but for most use cases, a FocusScope should probably prioritize the first tabbable to receive focus.

Reply:
I can make it so that the default behavior of `focusFirstInScope` is to first try tabbables, and if the walker finds no next node within the scope, move on to focusables.

Reply:
@LFDanLu See https://github.com/adobe/react-spectrum/commit/c69858c2b3b64d70957b1984b7c9a4e2a5fdf3d8